content/6_Advanced/Suffix_Array.mdx

```diff
@@ -35,7 +31,211 @@ frequency: 1
 	<Resource source="CP2" title="6.6 - Suffix Array and Applications"/>
 </Resources>
 
-### Implementations
+The suffix array of a string is the sorted array of all possible suffixes of the string.
+Each suffix is usually represented by its starting index.
+
+For example, if our string were `abcbcba`, the suffix array would be as follows:
+
+<center>
+
+| Suffix | String     |
+| ------ | ---------- |
+| 6      | `a`        |
+| 0      | `abcbcba`  |
+| 5      | `ba`       |
+| 3      | `bcba`     |
+| 1      | `bcbcba`   |
+| 4      | `cba`      |
+| 2      | `cbcba`    |
+
+</center>
+
+## Construction
+
+<FocusProblem problem="sample" />
+
+The general philosophy behind the construction of a suffix array is to sort incrementally.
+We first start by comparing the first character of each suffix, and then double the amount we compare until we're comparing the entire length of the string.
+If this sounds really abstract, don't worry!
+All the implementation details will be hammered out below.
+
+### Initial Transformation
+
+For convenience, let's start by appending a "null character" of sorts to the end of the string.
+This acts as a tiebreaker of sorts and will ensure we never hit the end of a suffix while comparing two suffixes.
```

Comment:
english teacher ahh comment but "of sorts" is repeated twice in close proximity 😅